pulp_ansible/app/tasks/collections.py

```diff
@@ -460,8 +474,9 @@ async def _pre_save(self, batch):
                 continue
 
             info = d_content.content.natural_key_dict()
+            deprecated = d_content.extra_data.get("deprecated", False)
             collection, created = Collection.objects.get_or_create(
-                namespace=info["namespace"], name=info["name"]
+                namespace=info["namespace"], name=info["name"], deprecated=deprecated
```

Comment:
This code is right, but this `_pre_save` stage only saves when the unit is created the first time. This means that a situation like

1. Pulp sync's a collection and deprecated=False
2. time passes and the deprecation flag is marked as True
3. This code will never execute to change deprecated=False because the unit is already in Pulp

The issue is that content is immutable in Pulp and this content type is expecting mutability.

Reply:
I can do a if..else to check if it is the first time creation

Reply:
This issue description is the first problem, it's signalling a deeper issue an if/else won't fix. For example, if we did get it "updating" an existing unit with the deprecated flag once changed, that would be changed for every repository everywhere, which versions are intended to be immutable.